pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java

```diff
@@ -1310,6 +1312,23 @@ public HierarchyTopicPolicies getHierarchyTopicPolicies() {
         return topicPolicies;
     }
 
+    @Override
+    public void registerTopicCacheCleanupFunction(TopicCacheCleanupFunction cleanupFunction) {
+        if (this.cleanupFunction != null) {
```

Comment:
do we have to enforce a concurrent access pattern ?
This method may be called from any threads

the same as for cleanupTopicCache

Reply:
That shouldn't happen. The logging will tell if that ever happens.